samply/src/mac/proc_maps.rs

```diff
@@ -227,8 +230,17 @@ fn enumerate_dyld_images(
         let (base_avma, image_file_path) = {
             let info_array_elem_addr =
                 info_array_addr + image_index as u64 * mem::size_of::<dyld_image_info>() as u64;
-            let image_info: &dyld_image_info =
-                unsafe { memory.get_type_ref_at_address(info_array_elem_addr) }?;
+            let mut image_info: MaybeUninit<dyld_image_info> = MaybeUninit::uninit();
+            let mut size = mem::size_of::<dyld_image_info>() as u64;
+            unsafe {
+                mach_vm_read_overwrite(task, info_array_elem_addr, size, &mut image_info as *mut MaybeUninit<dyld_image_info> as u64, &mut size).into_result()?;
+            }
+            
+            if size != mem::size_of::<dyld_image_info>() as u64 {
```

Comment:
We should also check that the result of `mach_vm_read_overwrite ` was `KERN_SUCCESS`, I think

Reply:
into_result does it